docs/Makefile.am

```diff
@@ -33,6 +33,21 @@ GENHTMLPAGES = curl.html curl-config.html mk-ca-bundle.html
 PDFPAGES = curl.pdf curl-config.pdf mk-ca-bundle.pdf
 MANDISTPAGES = curl.1.dist curl-config.1.dist
 
+packextrasdir=@docdir@
+# Docs installed to @docdir@ (i.e. /usr/local/share/doc/curl by default;
+# configurable by passing '--docdir' to './configure')
+packextras_DATA = \
+ BUGS.md \
+ CONTRIBUTE.md \
+ FAQ \
+ FEATURES.md \
+ INSTALL.md \
+ INTERNALS.md \
+ MANUAL.md \
+ README.md \
+ THANKS \
+ TODO
```

Comment:
Just curious, what's the litmus test to use for adding a documentation file to this list?

Reply:
A fair question. I started with the ones I saw in the Slackbuild.

The [docs installed by Debian](https://salsa.debian.org/debian/curl/-/blob/master/debian/libcurl4-doc.docs) are somewhat different but also somewhat similar.

```
README
docs/*.md
docs/FAQ
docs/KNOWN_BUGS
docs/THANKS
docs/TODO
```

AFAIK, the docs installed to docdir can be any docs the upstream maintainers think would be useful and relevant. You could even add a complete html version of your website, if I'm not mistaken.

Sometimes the INSTALL is included, sometimes not (because it's not needed once the package has been installed, and if there's an INSTALL file in docdir, that implies the package has been installed).

Reply:
It would be nice if there was a page that indicated conventions for docs, rather than me just writing my own beliefs here.

I did some searching and found this on a [Fedora packaging guide](https://docs.fedoraproject.org/en-US/packaging-guidelines/#_documentation)

>Any relevant documentation included in the source distribution should be included in the package in the proper documentation directory. Irrelevant documentation includes build instructions, the omnipresent INSTALL file containing generic build instructions, or example, and documentation for non-Linux systems, e.g. README.MSDOS. Also pay attention about which subpackage you include documentation in. For example API documentation belongs in the -devel subpackage, not the main one. Or if there’s a lot of documentation, consider putting it into a subpackage. In this case, it is recommended to use *-doc as the subpackage name.

Reply:
> I thought it would be a good idea to incorporate it here to make it a little easier for distro maintainers.

How this makes it easier for package maintainers is that if the docs go to the correct location using the upstream `make install`, then packagers don't have to add any special rules.

Example:

In this Slackbuild, I've specified the docdir during the configure (meson setup), and the docs will get installed to the tmp package directory.

[rmw Slackbuild](https://slackbuilds.org/slackbuilds/15.0/system/rmw/rmw.SlackBuild)

```
cd build
  CFLAGS="$SLKCFLAGS" \
  CXXFLAGS="$SLKCFLAGS" \
  meson .. \
    --buildtype=release \
    -Dstrip=true \
    -Db_sanitize=none \
    --mandir=/usr/man \
    --prefix=/usr \
    --localedir=/usr/share/locale \
    -Ddocdir=/usr/doc/$PRGNAM-$VERSION
  ninja
  DESTDIR=$PKG ninja install
cd ..
```

The package gets created soon after:

```
cd $PKG
/sbin/makepkg -l y -c n $OUTPUT/$PRGNAM-$VERSION-$ARCH-$BUILD$TAG.$PKGTYPE
```

Reply:
in that case, then I think all docs should be installed except those that are specifically about specific internal details (like BUGREF, CURL_DISABLE and DYNBUF).